lintly/parsers.py

```diff
@@ -607,6 +607,68 @@ def parse_violations(self, output):
         return violations
 
 
+class SemgrepParser(BaseLintParser):
+    """
+    Semgrep JSON format
+{
+    "errors": [],
+    "results": [{
+        "check_id": "python.jinja2.security.audit.autoescape-disabled.autoescape-disabled",
+        "end": {
+            "col": 77,
+            "line": 14
+        },
+        "extra": {
+            "fix_regex": {
+                "regex": "(.*)\\)",
+                "replacement": "\\1, autoescape=True)"
+            },
+            "is_ignored": false,
+            "lines": "env = Environment(loader=FileSystemLoader(TEMPLATES_PATH), autoescape=False)",
+            "message": "Detected a Jinja2 environment without autoescaping.",
+            "metadata": {
+                "category": "security",
+                "cwe": "CWE-116: Improper Encoding or Escaping of Output",
+                "license": "Commons Clause License Condition v1.0[LGPL-2.1-only]",
+                "owasp": "A6: Security Misconfiguration",
+                "references": ["https://jinja.palletsprojects.com/en/2.11.x/api/#basics"],
+                "source": "https://semgrep.dev/r/python.jinja2.security.audit.autoescape-disabled.autoescape-disabled",
+                "technology": ["jinja2"]
+            },
+            "metavars": {},
+            "severity": "WARNING"
+        },
+        "path": "path/to/file",
+        "start": {
+            "col": 7,
+            "line": 14
+        }
+    }]
+}
+    """
+
+    def parse_violations(self, output):
+        if not output:
+            return dict()
+
+        json_data = json.loads(output)
+
+        violations = collections.defaultdict(list)
+
+        for violation_json in json_data["results"]:
```

Comment:
Didn't you say you were going to report only on violations where `violation_json["extra"]["metadata"]["category"]` is `"security"`? Did you end up walking back on that decision?

Reply:
So initial thought was that when I was using generic config and then I found specific to `security` ruleset which is this one `p/r2c-security-audit` where it all security category issues. So no need to filter it out.